library/alloc/src/vec.rs

```diff
@@ -2928,6 +2934,35 @@ impl<T> IntoIter<T> {
         self.ptr = self.buf.as_ptr();
         self.end = self.buf.as_ptr();
     }
+
+    /// Shifts the remaining elements to `offset` and then converts the whole allocation into a Vec
+    /// with `vec.len() == offset + self.len()`
+    ///
+    /// # Safety
+    ///
+    /// When a non-zero offset is passed the resulting Vec will have an uninitialized prefix
+    /// that needs to be filled before the Vec is valid again.
+    ///
+    /// * `offset + self.len()` must not exceed `self.cap`
+    /// * `offset == 0` is always valid
+    /// * `offset` must be positive
+    unsafe fn into_vec(self, offset: isize) -> Vec<T> {
+        let dst = unsafe { self.buf.as_ptr().offset(offset) };
```

Comment:
We could probably use `add` instead of `offset` for `usize`, but I don't know if the compiler will optimize out the `offset` if it is `0`? Does it or should we have two functions, one with offset and one without?

Reply:
0 is a constant in the other callsite, so it should be easy to optimize for llvm.

Reply:
cc @lzutao to see if he's interested to check this out